Mhmd/dasbhoard

Mhmd0Mhmod opened the pull request:
This pull request introduces significant updates to the dashboard application, including new feature implementations, UI enhancements, and codebase restructuring. The changes focus on improving the user interface, enhancing modularity, and replacing legacy implementations with modern components.

### UI Enhancements:
* Redesigned the `CampaignsPage` and `DashboardPage` to include skeleton loaders, suspense components, and improved layouts for better user experience (`app/dashboard/campaigns/page.tsx`, `app/dashboard/page.tsx`). [[1]](diffhunk://#diff-2a6824a7b93b559a8c591d3276978642e62b11ecce1574e46d9dae6493e327cdL12-R75) [[2]](diffhunk://#diff-e16c1c1133d5d72341ada507f3e06fa37dd79de574403f158159ff934f9b5013L19-R83)
* Added new reusable components, including `TableSkeleton`, `DatePicker`, and `Icon`, for improved modularity and maintainability (`components/TableSkeleton.tsx`, `components/DatePicker.tsx`, `components/Icon.tsx`). [[1]](diffhunk://#diff-4283b6885353a5daae4ecd74cda93910fa1b5d9512cd64eaae89137dcc544d19R1-R86) [[2]](diffhunk://#diff-c9618c573214934cdb6d57610674ef817a25c0bfb2d441061d8ef2d56c91555cR1-R45) [[3]](diffhunk://#diff-1a9430e87526ba32d5666a8f555688d8bc38fd697a99a82468f7c635b96a0abdR1-R13)

### Codebase Restructuring:
* Refactored the `CampaignCreatePage` to use modern UI components and context providers, replacing the previous implementation (`app/dashboard/campaigns/create/page.tsx`, `app/dashboard/campaigns/(.)create/page.tsx`). [[1]](diffhunk://#diff-22093aa7f4e5d135b48432070885afb5c118f0486d2d2091eff3a8a0cdb430d3L1-R35) [[2]](diffhunk://#diff-5243ad3f2c741174b6b9e2c0df56c8b8b2cd21e36aa43d70240d5562039e8c06R1-R37)
* Updated the `KpiCard` component to support icons and colors, simplifying its API and improving visual clarity (`components/KpiCard.tsx`).


These changes collectively improve the maintainability, performance, and user experience of the dashboard application.

Video : https://www.loom.com/share/39265ce044444f9b9a35550eec37b4ce?sid=83503ef3-3d38-4217-a7a0-f83be4020fed
<img width="1920" height="1413" alt="image" src="https://github.com/user-attachments/assets/c6d16734-b8a8-43d8-a619-3291397fcb25" />
<img width="1920" height="878" alt="image" src="https://github.com/user-attachments/assets/b16cd179-dc36-4c20-9793-f383f493e0c0" />

Mhmd0Mhmod committed:
feat: update sidebar navigation paths and add new dashboard pages for analytics, leads, and onboarding

Mhmd0Mhmod committed:
fix: correct dashboard link path in sidebar navigation

Mhmd0Mhmod committed:
feat: enhance dashboard layout with new components and loading states

Mhmd0Mhmod committed:
feat: rename page components for consistency and improve layout structure

Mhmd0Mhmod committed:
feat: refactor sidebar navigation components and improve link handling

Mhmd0Mhmod committed:
Improve SidebarLink styles and formatting

Refactored SidebarLink.tsx for better code formatting and enhanced button styling, including improved hover effects and color transitions for active and highlighted states. Also added minor whitespace and formatting adjustments for consistency.

Mhmd0Mhmod committed:
feat: enhance DashboardRootLayout with improved SidebarInset styling and structure

Mhmd0Mhmod committed:
feat: clean up .idea directory by removing unnecessary files and updating .gitignore

Mhmd0Mhmod committed:
feat: add HelpPage and CampaignsFilter components

Mhmd0Mhmod committed:
feat: remove KpiCardSkeleton component and update KpiCards to apply color styles directly

fix: enhance Calendar component with improved class names and button variants

feat: add DatePicker component for date selection with a calendar popover

feat: implement TableSkeleton component for loading state representation in tables

feat: create CampaignsTable component to display campaign data with actions

feat: add StatsCardSkeleton component for loading state representation in stats cards

feat: implement StatsCards component to display various campaign statistics

feat: add Calendar01 component for showcasing the calendar functionality

Mhmd0Mhmod committed:
feat: implement campaign creation flow with multi-step form and context management

Mhmd0Mhmod committed:
Refactor campaign creation flow with multi-step form and validation

- Introduced NavigationButtons component for step navigation.
- Updated CampaignCreatePage to use Card components for better UI structure.
- Implemented individual step components: CampaignDetailsStep, LeadsSelectionStep, MailboxAssignmentStep, SequenceBuilderStep, ReviewLaunchStep, and ScheduleSettingStep.
- Added Zod validation schema for campaign form data.
- Removed ScheduleConfigurationStep as it was redundant.
- Enhanced loading state UI for better user experience.

Mhmd0Mhmod committed:
feat: implement dialog context and refactor campaign creation page with dialog components

Mhmd0Mhmod committed:
feat: refactor campaign creation and editing pages with new dialog components and context management

Mhmd0Mhmod committed:
feat: implement campaign edit and create pages with dialog components and skeleton loading

Mhmd0Mhmod committed:
feat: Refactor Campaign Skeleton and Dialog Context, add Campaign Actions and Data Components

- Updated CampaignSkeleton component to remove DialogHeader and DialogFooter, replacing them with a simpler layout.
- Enhanced DialogContextProvider to accept an optional className prop for better styling flexibility.
- Introduced statusCampaign enum in campaign types for better status management.
- Created CampaignsActions component to handle various campaign actions (view, edit, delete, pause, resume, copy).
- Implemented CampaignsTableRow component to display campaign details and actions in a table format.
- Developed ActivityTab, CampaignHeader, CampiagnTabs, LeadsTab, SequenceTab, and StatsTab components for campaign data visualization.
- Added mock data for campaigns, sequence steps, leads, and activity logs to support new components.

Mhmd0Mhmod committed:
feat: remove ActivityTab component and associated tabs, add LeadsFilter for enhanced lead management

Mhmd0Mhmod committed:
feat: add notFound handling in CampaignCreatePage for better error management

Mhmd0Mhmod committed:
feat: enhance campaign detail and statistics components with improved UI and functionality

Mhmd0Mhmod committed:
feat: refactor campaign tabs and steps, add lead and mailbox data sources

Israel-Laguan commented on components/campaigns/campaignData/StatsTab.tsx:
```diff
@@ -0,0 +1,322 @@
+}
+
+const generateData = (days: number): ChartData[] => {
+  const data: ChartData[] = [];
+  const today = new Date();
+  
+  for (let i = days - 1; i >= 0; i--) {
+    const date = new Date(today);
+    date.setDate(today.getDate() - i);
+    
+    const sent = Math.floor(Math.random() * 50) + 15;
+    const opened = Math.floor(sent * (0.25 + Math.random() * 0.35));
+    const clicked = Math.floor(opened * (0.15 + Math.random() * 0.30));
+    const replied = Math.floor(opened * (0.10 + Math.random() * 0.20));
+    const bounced = Math.floor(sent * (0.02 + Math.random() * 0.08));
+    
+    data.push({
+      date: date.toISOString().split('T')[0],
+      sent,
+      opened,
+      replied,
+      bounced,
+      clicked,
+      formattedDate: date.toLocaleDateString('en-US', { 
+        day: 'numeric', 
+        month: 'short' 
+      }),
+    });
+  }
+  
+  return data;
+};
+
```
Move to an adjacent file like "stats.mock.ts" and import

Israel-Laguan commented on components/campaigns/CampaignsFilter.tsx:
```diff
@@ -0,0 +1,150 @@
+import { Calendar, CheckIcon, Search, Server } from "lucide-react";
+import { useState } from "react";
+import DatePicker from "../DatePicker";
+const availableMailboxes = [
+  "john@mycompany.com",
+  "sarah@mycompany.com",
+  "mike@mycompany.com",
+  "lisa@mycompany.com",
+  "david@mycompany.com",
+];
```
Move to a mock like 

```
const domains = {
'mycompany.com': [
  "john@mycompany.com",
  "sarah@mycompany.com",
  "mike@mycompany.com",
  "lisa@mycompany.com",
  "david@mycompany.com",
]
}
```

Or a similar structure

Israel-Laguan replied:
Chek if you can reuse the data from [`lib/data/campaigns.ts`](lib/data/campaigns.ts)

Israel-Laguan commented on components/dashboard/KpiCards.tsx:
```diff
@@ -0,0 +1,34 @@
+import { Mail, Send, TrendingUp, Users } from "lucide-react";
+import KpiCard from "../KpiCard";
+const statsCards = [
+  { title: "Active Campaigns", value: "12", icon: Send, color: "bg-blue-500" },
+  {
+    title: "Leads Contacted",
+    value: "2,847",
+    icon: Users,
+    color: "bg-green-500",
+  },
+  { title: "Open Rate", value: "34.2%", icon: Mail, color: "bg-purple-500" },
+  {
+    title: "Reply Rate",
+    value: "8.7%",
+    icon: TrendingUp,
+    color: "bg-orange-500",
+  },
+];
```
Try to reuse [`lib/data/campaigns.ts`](lib/data/campaigns.ts) or move to a mock

Israel-Laguan commented on components/dashboard/RecentReply/RecentReplyList.tsx:
```diff
@@ -0,0 +1,74 @@
+const recentReplies = [
+  {
+    name: "Sarah Johnson",
+    email: "sarah@techcorp.com",
+    company: "TechCorp",
+    message:
+        "Thanks for reaching out! I'd love to schedule a call to discuss this further.",
+    time: "2 hours ago",
+    type: "positive",
+  },
+  {
+    name: "Mike Chen",
+    email: "mike@startup.io",
+    company: "Startup.io",
+    message:
+        "Not interested at this time, but please keep us in mind for the future.",
+    time: "4 hours ago",
+    type: "neutral",
+  },
+  {
+    name: "Lisa Rodriguez",
+    email: "lisa@enterprise.com",
+    company: "Enterprise Inc",
+    message:
+        "This looks interesting. Can you send me more information about pricing?",
+    time: "6 hours ago",
+    type: "positive",
+  },
+];
```
Move to a mock

Israel-Laguan replied:
Check if you can reuse [`lib/data/campaigns.ts`](lib/data/campaigns.ts), for example `activityLog`

Israel-Laguan commented on components/dashboard/compaigns/StatsCards.tsx:
```diff
@@ -0,0 +1,51 @@
+import KpiCard from "@/components/KpiCard";
+import { Eye, Mail, Send, TrendingUp, Users } from "lucide-react";
+const statsCards = [
+  {
+    title: "Total Compaigns",
+    value: "12",
+    icon: Send,
+    color: "bg-blue-100 text-blue-600",
+  },
+  {
+    title: "Total Sent",
+    value: "2,847",
+    icon: Mail,
+    color: "text-purple-600 bg-purple-100",
+  },
+  {
+    title: "Total Replies",
+    value: "8.7%",
+    icon: TrendingUp,
+    color: "bg-green-100 text-green-600",
+  },
+  {
+    title: "Open Rate",
+    value: "34.2%",
+    icon: Eye,
+    color: "text-orange-500  bg-orange-100",
+  },
+  {
+    title: "Reply Rate",
+    value: "8.7%",
+    icon: Users,
+    color: "text-pink-600 bg-pink-100",
+  },
+];
```
Move to a mock

Israel-Laguan replied:
Check if you can reuse the mock in [`lib/data/campaigns.ts`](lib/data/campaigns.ts)

Israel-Laguan commented on lib/data/campaigns.ts:
```diff
@@ -0,0 +1,188 @@
+  },
+];
+
+export const campaignLeads = [
```
Maybe unify with [`lib/data/leads.ts`](lib/data/leads.ts)

Israel-Laguan commented on lib/data/campaigns.ts:
```diff
@@ -0,0 +1,188 @@
+];
+
+
+export type Campaign = {
+  id: number;
+  name: string;
+  status: statusCampaign; 
+  mailboxes: number;
+  leadsSent: number;
+  replies: number;
+  openRate: string;
+  replyRate: string;
+  lastSent: string;
+  createdDate: string;
+  assignedMailboxes: string[];
+};
```
This one maybe move to a central type file instead

Mhmd0Mhmod replied:
keep it here until finish because that's will case a problem in build 
the old components use old campaign interface in types file called `campaign.ts` in `@/types`

Israel-Laguan commented on components/calendar-01.tsx:
Don't forget to delete this component later

Israel-Laguan commented on app/dashboard/page.tsx:
```diff
@@ -16,7 +23,62 @@ export default async function DashboardPage() {
+  // const dashboardData = await getDashboardMockDataAction("companyId");
 
-  return <DashboardContent dashboardData={dashboardData} />;
+  // return <DashboardContent dashboardData={dashboardData} />;
```
Seems like currently you moved the data in the lower components, maybe would be better to get the mock data here and pass down the data to the children components

Mhmd0Mhmod committed:
feat: refactor dashboard components for improved structure and performance, integrate new data sources

Mhmd0Mhmod committed:
feat: refactor campaign creation and editing components, enhance navigation logic and context management

Mhmd0Mhmod committed:
feat: simplify NavigationButtons component by removing unused imports and optimizing canProceed function

Mhmd0Mhmod committed:
feat: enhance campaign steps with timezone selection, email limits, and sequence management

Mhmd0Mhmod committed:
feat: remove unused campaign components and update link path for new campaign

Israel-Laguan commented on app/dashboard/page.tsx:
```diff
@@ -16,7 +15,7 @@ export default async function DashboardPage() {
   // }
 
-  const dashboardData = await getDashboardMockDataAction("companyId");
+  // const dashboardData = await getDashboardMockDataAction("companyId");
```
Remove these comments if not needed in the future (as each component would be responsible to grab its own data)

Mhmd0Mhmod committed:
Remove unused authentication code from dashboard

Commented-out authentication and session management code was removed from DashboardPage for clarity and to clean up the file. The page now simply renders DashboardContent.

Mhmd0Mhmod committed:
feat: remove unused imports and actions from LeadsTab component

Mhmd0Mhmod committed:
feat: update LeadsSelectionStep to handle leads list changes and add leadsList data to campaigns

Mhmd0Mhmod committed:
Merge branch 'dev' into mhmd/dasbhoard

Mhmd0Mhmod committed:
fix: add missing closing brace to DashboardPage function
